networkx/classes/tests/dispatch_interface.py

```diff
@@ -49,25 +49,112 @@ def convert(graph):
 
 
 class LoopbackDispatcher:
-    non_toplevel = {
-        "inter_community_edges": nx.community.quality.inter_community_edges,
-        "is_tournament": nx.algorithms.tournament.is_tournament,
-        "mutual_weight": nx.algorithms.structuralholes.mutual_weight,
-        "score_sequence": nx.algorithms.tournament.score_sequence,
-        "tournament_matrix": nx.algorithms.tournament.tournament_matrix,
-    }
-
     def __getattr__(self, item):
-        # Return the original, undecorated NetworkX algorithm
-        if hasattr(nx, item):
-            return getattr(nx, item)._orig_func
-        if item in self.non_toplevel:
-            return self.non_toplevel[item]._orig_func
-        raise AttributeError(item)
+        try:
+            return nx.utils.backends._registered_algorithms[item].__wrapped__
+        except KeyError:
+            raise AttributeError(item) from None
 
     @staticmethod
-    def convert_from_nx(graph, weight=None, *, name=None):
-        return graph
+    def convert_from_nx(
+        graph,
+        *,
+        edge_attrs=None,
+        node_attrs=None,
+        preserve_edge_attrs=None,
+        preserve_node_attrs=None,
+        preserve_graph_attrs=None,
+        name=None,
+        graph_name=None,
```

Comment:
Including `graph_name` in `convert_from_nx` greatly improved my debugging and development experience when working on this PR.